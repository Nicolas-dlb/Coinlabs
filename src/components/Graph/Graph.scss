.graph_container {
	.graph {
		position: relative;
		top: 4vh;

		display: flex;
		flex-direction: column;
		align-items: center;
		justify-content: center;
		.recharts-responsive-container {
			position: relative;
			top: 2vh;
		}
		width: 100%;
		.tooltip-top {
			display: flex;
			align-items: center;
			justify-content: space-between;
			svg {
				width: 11px;
			}
		}
		.tooltip-bottom {
			display: flex;
			align-items: center;
			font-size: 21px;
			margin-top: 15px;
			margin-bottom: 2px;
		}
		&::after {
			content: "";
			position: absolute;
			top: -4vh;
			left: 0;
			right: 5.2%;
			bottom: 70.2vh;
			border-radius: 20px;
			-webkit-box-shadow: inset 0px 13px 31px 39px rgba(34, 30, 52, 1);
			-moz-box-shadow: inset 0px 13px 31px 39px rgba(34, 30, 52, 1);
			box-shadow: inset 0px 13px 31px 39px rgba(34, 30, 52, 1);

			pointer-events: none;
			z-index: 1;
		}
	}
}
