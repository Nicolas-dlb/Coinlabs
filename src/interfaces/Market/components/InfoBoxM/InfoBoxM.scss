.infoboxM {
	border-radius: 12px;
	background-color: var(--third-bg-color);
	width: 250px;
	height: 7vh;
	display: flex;
	align-items: center;
	justify-content: center;
	padding: 20px;
	margin: 8px;
	padding-right: 29px;

	transition: all 0.5s ease;
	transition: background-color 0.3s ease;
	.graph {
		&::after {
			display: none;
		}
	}

	.fix_ethereum {
		margin-left: 10px;
		padding-right: 15px !important;
	}
	.infobox_left {
		display: flex;
		flex-direction: column;
		align-items: flex-start;
		justify-content: center;
		width: fit-content;
		padding-right: 25px;

		.infobox_left_top {
			display: flex;
			width: fit-content;
			align-items: center;
			justify-content: flex-start;

			p {
				color: #c4c3c9;
				margin-right: 10px;
			}
			svg {
				height: 15px;
			}
		}
		.infobox_left_bottom {
			font-size: 20px;
			width: 100px;
			display: flex;
			align-items: center;
			justify-content: flex-start;
			position: relative;
			top: 1.3vh;
			p:first-child {
				margin-right: 5px;
			}
		}
	}
	.infobox_right {
		width: 15vw;
		padding-left: 35px;

		.graph {
			.recharts-responsive-container {
				position: relative;
				top: 25px;
				right: 12px;
			}
		}
	}
}

@media only screen and (min-width: 992px) {
	.infoboxM {
		padding-right: 0px;
		width: 225px;
		padding-left: 25px;
		&:first-child {
			margin-left: 0;
		}
		.infobox_right {
			padding-left: 0px;
		}
	}
}
