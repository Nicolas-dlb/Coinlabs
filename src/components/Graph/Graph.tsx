import {
  AreaChart,
  Area,
  XAxis,
  YAxis,
  Tooltip,
  ResponsiveContainer,
} from "recharts";
import "./Graph.scss";
import React, { useEffect, useState } from "react";
import { getNumberFixed, numberWithSpaces } from "utils/utils";
import Grid from "./Grid/Grid";

type GraphProps = {
  data: Array<Object>;
  color: string;
  width: number;
  height: number | undefined;
  tooltip?: true;
  selected?: string;
};

function Graph({
  data,
  color,
  width,
  height,
  tooltip = undefined,
  selected = "",
}: GraphProps) {
  const [number, setNumber] = useState(0);

  useEffect(() => {
    if (selected === "Bitcoin" || selected === "All") {
      setNumber(0);
    } else if (selected === "Ethereum") {
      setNumber(1);
    } else if (selected === "Litecoin") {
      setNumber(2);
    } else if (selected === "Ripple") {
      setNumber(4);
    } else if (selected === "Neo") {
      setNumber(3);
    }
  }, [selected]);

  const CustomTooltip = ({ active, payload, num }: any) => {
    let angle;
    if (payload) {
      angle = payload[0]?.value > 0 ? "140deg" : "215deg";
    }
    const divise =
      num === 0
        ? 1
        : num === 1
        ? 11
        : num === 2
        ? 100
        : num === 4
        ? 5000
        : num === 3
        ? 200
        : 1;
    const price = payload[num]?.value / divise;
    if (active) {
      return (
        <>
          <div className="tooltip-top">
            <p style={{ marginRight: "25px" }}>Balance</p>
            <svg
              style={{ transform: `rotate(${angle})` }}
              xmlns="http://www.w3.org/2000/svg"
              viewBox="0 0 492 492"
              fill="#fff"
            >
              <path d="M464.3 207.4l0.8 0.2H135.9l103.5-103.7c5.1-5.1 7.8-11.9 7.8-19.1 0-7.2-2.8-14-7.8-19.1L223.3 49.5c-5.1-5.1-11.8-7.9-19-7.9 -7.2 0-14 2.8-19 7.8L7.8 226.9C2.8 232 0 238.8 0 246c0 7.2 2.8 14 7.8 19.1l177.4 177.4c5.1 5.1 11.8 7.8 19 7.8 7.2 0 13.9-2.8 19-7.8l16.1-16.1c5.1-5.1 7.8-11.8 7.8-19 0-7.2-2.8-13.6-7.8-18.7L134.7 284.4h330c14.8 0 27.3-12.8 27.3-27.6v-22.8C492 219.2 479.2 207.4 464.3 207.4z" />
            </svg>
            <p style={{ marginLeft: "6px" }}>12%</p>
          </div>
          <div className="tooltip-bottom">
            <p>$</p>
            <p className="tooltip_price" id="tooltip_price">
              {numberWithSpaces(getNumberFixed(price, 2))}
            </p>
          </div>
        </>
      );
    }

    return null;
  };

  const wrapperStyle = {
    borderRadius: "15px",
    background:
      number === 0
        ? "linear-gradient(20deg, rgba(11,171,81,1) 0%, rgba(30,171,155,1) 100%)"
        : number === 1
        ? "linear-gradient(292deg, rgba(105,162,200,1) 0%, rgba(72,140,185,1) 100%)"
        : number === 2
        ? "linear-gradient(233deg, rgba(203,161,128,1) 0%, rgba(219,154,103,1) 100%)"
        : number === 4
        ? "linear-gradient(225deg, rgba(170,134,190,1) 0%, rgba(160,117,185,1) 30%)"
        : "linear-gradient(233deg, rgba(117,174,218,1) 0%, rgba(74,156,192,1) 40%)",
    border: "none",
    color: "#fff",
    padding: "10px",
    paddingTop: "12px",
    BoxShadow: "0px 6px 18px -6px rgba(0,0,0,0.49)",
    WebkitBoxShadow: "0px 6px 18px -6px rgba(0,0,0,0.49)",
    MozBoxShadow: "0px 6px 18px -6px rgba(0,0,0,0.49)",
    alignItems: "space-evenly",
    justifyContent: "center",
    zIndex: 1000,
  };

  return (
    <div className="graph_container">
      <div className="graph">
        <ResponsiveContainer
          width={width || (!width && tooltip ? "90%" : "99%")}
          height={height}
        >
          <AreaChart width={width} height={height} data={data}>
            <defs>
              <linearGradient id="Green" x1="1" y1="0" x2="0" y2="1">
                <stop offset="0%" stopColor="#2aac63" stopOpacity={0.17} />
                <stop offset="100%" stopColor="#2aac63" stopOpacity={0} />
              </linearGradient>
              <linearGradient id="White" x1="0" y1="0" x2="0" y2="1">
                <stop offset="0%" stopColor="#fff" stopOpacity={0} />
                <stop offset="100%" stopColor="#fff" stopOpacity={0} />
              </linearGradient>
              <linearGradient id="Blue" x1="1" y1="0" x2="0" y2="1">
                <stop offset="0%" stopColor="#4689b6" stopOpacity={0.17} />
                <stop offset="100%" stopColor="#4689b6" stopOpacity={0} />
              </linearGradient>
              <linearGradient id="Pink" x1="1" y1="0" x2="0" y2="1">
                <stop offset="0%" stopColor="#a075b9" stopOpacity={0.17} />
                <stop offset="100%" stopColor="#a075b9" stopOpacity={0} />
              </linearGradient>
              <linearGradient id="Orange" x1="1" y1="0" x2="0" y2="1">
                <stop offset="0%" stopColor="#db9a67" stopOpacity={0.17} />
                <stop offset="100%" stopColor="#db9a67" stopOpacity={0} />
              </linearGradient>
              <linearGradient id="Blue2" x1="1" y1="0" x2="0" y2="1">
                <stop offset="0%" stopColor="#4a9cc0" stopOpacity={0.17} />
                <stop offset="100%" stopColor="#4a9cc0" stopOpacity={0} />
              </linearGradient>
            </defs>
            <XAxis
              type="number"
              dataKey="tokens"
              hide
              interval="preserveStartEnd"
              tickMargin={25}
              tickLine
              axisLine={false}
              tickCount={5}
            />
            <YAxis hide tickMargin={25} tickLine axisLine={false} />
            {tooltip && (
              <Tooltip
                content={<CustomTooltip num={number} />}
                wrapperStyle={wrapperStyle}
                itemStyle={{ color: "rgb(11,171,81)" }}
                cursor={{
                  stroke:
                    number === 0
                      ? "rgb(11,171,81)"
                      : number === 1
                      ? "#4689b6"
                      : number === 2
                      ? "#db9a67"
                      : number === 4
                      ? "#a075b9"
                      : "#4a9cc0",
                  strokeWidth: 1.8,
                  strokeDasharray: 10,
                }}
                labelFormatter={(value: any) => `Tokens: ${value}`}
              />
            )}
            <Area
              activeDot={{
                stroke: "#fff",
                fill: "#26ae5f",
                strokeWidth: number === 0 ? 3 : 0,
                r: number === 0 ? 5 : 0,
              }}
              onMouseOver={() => {
                setNumber(0);
              }}
              strokeWidth={
                selected.includes("Bitcoin") ||
                selected.includes("All") ||
                selected === ""
                  ? 1
                  : 0
              }
              type="monotone"
              dataKey="Price"
              stroke={
                color === "Red"
                  ? "#f0487c"
                  : color === "Green"
                  ? "#2aac63"
                  : color === "White"
                  ? "#fff"
                  : "#fff"
              }
              fillOpacity={1}
              fill={
                selected.includes("Bitcoin") ||
                selected.includes("All") ||
                selected === ""
                  ? `url(#${color})`
                  : "none"
              }
            />

            <Area
              activeDot={{
                stroke: "#fff",
                fill: "#4689b6",
                strokeWidth: number === 1 ? 3 : 0,
                r: number === 1 ? 5 : 0,
              }}
              onMouseOver={() => {
                setNumber(1);
              }}
              strokeWidth={
                selected.includes("Ethereum") || selected.includes("All")
                  ? 1
                  : 0
              }
              type="monotone"
              dataKey="ethereum"
              stroke="#4689b6"
              fillOpacity={1}
              fill={
                selected.includes("Ethereum") || selected.includes("All")
                  ? `url(#Blue)`
                  : "none"
              }
            />
            <Area
              activeDot={{
                stroke: "#fff",
                fill: "#db9a67",
                strokeWidth: number === 2 ? 3 : 0,
                r: number === 2 ? 5 : 0,
              }}
              onMouseOver={() => {
                setNumber(2);
              }}
              strokeWidth={
                selected.includes("Litecoin") || selected.includes("All")
                  ? 1
                  : 0
              }
              type="monotone"
              dataKey="litecoin"
              stroke="#db9a67"
              fillOpacity={1}
              fill={
                selected.includes("Litecoin") || selected.includes("All")
                  ? `url(#Orange)`
                  : "none"
              }
            />

            <Area
              activeDot={{
                stroke: "#fff",
                fill: "#4a9cc0",
                strokeWidth: number === 3 ? 3 : 0,
                r: number === 3 ? 5 : 0,
              }}
              onMouseOver={() => {
                setNumber(3);
              }}
              strokeWidth={
                selected.includes("Neo") || selected.includes("All") ? 1 : 0
              }
              type="monotone"
              dataKey="neo"
              stroke="#4a9cc0"
              fillOpacity={1}
              fill={
                selected.includes("Neo") || selected.includes("All")
                  ? `url(#Blue2)`
                  : "none"
              }
            />

            <Area
              activeDot={{
                stroke: "#fff",
                fill: "#a075b9",
                strokeWidth: number === 4 ? 3 : 0,
                r: number === 4 ? 5 : 0,
              }}
              onMouseOver={() => {
                setNumber(4);
              }}
              strokeWidth={
                selected.includes("Ripple") || selected.includes("All") ? 1 : 0
              }
              type="monotone"
              dataKey="ripple"
              stroke="#a075b9"
              fillOpacity={1}
              fill={
                selected.includes("Ripple") || selected.includes("All")
                  ? `url(#Pink)`
                  : "none"
              }
            />
          </AreaChart>
        </ResponsiveContainer>
        <Grid />
      </div>
    </div>
  );
}

export default Graph;
