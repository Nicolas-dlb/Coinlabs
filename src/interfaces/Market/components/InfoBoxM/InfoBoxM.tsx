import "./InfoBoxM.scss";
import { numberWithSpaces } from "utils/utils";
import React from "react";
import Graph from "components/Graph2/Graph";

type InfoBoxProps = {
  name: string;
  price: number;
  variation: number;
  data: any;
};
function InfoBox({ name, price, variation, data }: InfoBoxProps) {
  const color = variation && variation > 0 ? "#37bb77" : "#f1c94f";
  const angle = variation && variation > 0 ? "140deg" : "210deg";

  let timelineData = [...Array(110).keys()].map((x) => ({
    tokens: x,
    Price: 0,
    timestamp: 37364728,
  }));
  if (data) {
    timelineData = [...Array(data?.length).keys()].map((x) => ({
      tokens: x,
      Price: data[x][1],
      timestamp: data[x][0],
    }));
  }
  return (
    <div className="infoboxM">
      <div
        className={
          name === "Ethereum" ? "infobox_left fix_ethereum" : "infobox_left"
        }
      >
        <div className="infobox_left_top">
          <p id="infoBoxM_name">{name}</p>
          <svg
            style={{ transform: `rotate(${angle})` }}
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 492 492"
            fill={color}
          >
            <path d="M464.3 207.4l0.8 0.2H135.9l103.5-103.7c5.1-5.1 7.8-11.9 7.8-19.1 0-7.2-2.8-14-7.8-19.1L223.3 49.5c-5.1-5.1-11.8-7.9-19-7.9 -7.2 0-14 2.8-19 7.8L7.8 226.9C2.8 232 0 238.8 0 246c0 7.2 2.8 14 7.8 19.1l177.4 177.4c5.1 5.1 11.8 7.8 19 7.8 7.2 0 13.9-2.8 19-7.8l16.1-16.1c5.1-5.1 7.8-11.8 7.8-19 0-7.2-2.8-13.6-7.8-18.7L134.7 284.4h330c14.8 0 27.3-12.8 27.3-27.6v-22.8C492 219.2 479.2 207.4 464.3 207.4z" />
          </svg>
        </div>
        <div className="infobox_left_bottom">
          <p>$</p>
          <p id="infoboxM_price">{numberWithSpaces(price)}</p>
        </div>
      </div>
      <div className="infobox_right">
        <Graph data={timelineData} color="White" width={115} height={90} />
      </div>
    </div>
  );
}

export default InfoBox;
