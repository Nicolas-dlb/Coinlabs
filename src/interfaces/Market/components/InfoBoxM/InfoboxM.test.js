import { mount } from "enzyme";
import React from "react";
import { act } from "react-dom/test-utils";
import Graph from "components/Graph/Graph";
import InfoBoxM from "./InfoBoxM";

describe("InfoBoxM", () => {
  let InfoBoxMComponent;
  beforeEach(async () => {
    await act(async () => {
      InfoBoxMComponent = mount(
        <InfoBoxM
          name="Bitcoin"
          price={25}
          variation={10}
          data={[{ tokens: 0, price: 0, timestamp: 0 }]}
        />
      );
    });
  });

  it("should render correctly", () => {
    expect(InfoBoxMComponent).toMatchSnapshot();
  });
  it("should render graph component", () => {
    expect(InfoBoxMComponent.containsMatchingElement(<Graph />)).toBe(true);
  });
  it("should render custom price", () => {
    expect(InfoBoxMComponent.find("#infoboxM_price").text()).toBe("25");
  });

  it("should render custom name", () => {
    expect(InfoBoxMComponent.find("#infoBoxM_name").text()).toBe("Bitcoin");
  });
});
