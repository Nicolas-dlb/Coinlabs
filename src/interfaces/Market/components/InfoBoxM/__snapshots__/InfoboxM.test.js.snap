// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBoxM should render correctly 1`] = `
<InfoBox
  data={
    Array [
      Object {
        "price": 0,
        "timestamp": 0,
        "tokens": 0,
      },
    ]
  }
  name="Bitcoin"
  price={25}
  variation={10}
>
  <div
    className="infoboxM"
  >
    <div
      className="infobox_left"
    >
      <div
        className="infobox_left_top"
      >
        <p
          id="infoBoxM_name"
        >
          Bitcoin
        </p>
        <svg
          fill="#37bb77"
          style={
            Object {
              "transform": "rotate(140deg)",
            }
          }
          viewBox="0 0 492 492"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464.3 207.4l0.8 0.2H135.9l103.5-103.7c5.1-5.1 7.8-11.9 7.8-19.1 0-7.2-2.8-14-7.8-19.1L223.3 49.5c-5.1-5.1-11.8-7.9-19-7.9 -7.2 0-14 2.8-19 7.8L7.8 226.9C2.8 232 0 238.8 0 246c0 7.2 2.8 14 7.8 19.1l177.4 177.4c5.1 5.1 11.8 7.8 19 7.8 7.2 0 13.9-2.8 19-7.8l16.1-16.1c5.1-5.1 7.8-11.8 7.8-19 0-7.2-2.8-13.6-7.8-18.7L134.7 284.4h330c14.8 0 27.3-12.8 27.3-27.6v-22.8C492 219.2 479.2 207.4 464.3 207.4z"
          />
        </svg>
      </div>
      <div
        className="infobox_left_bottom"
      >
        <p>
          $
        </p>
        <p
          id="infoboxM_price"
        >
          25
        </p>
      </div>
    </div>
    <div
      className="infobox_right"
    >
      <Graph
        color="White"
        data={
          Array [
            Object {
              "Price": undefined,
              "timestamp": undefined,
              "tokens": 0,
            },
          ]
        }
        height={90}
        width={115}
      >
        <div
          className="graph_container"
        >
          <div
            className="graph"
          >
            <ResponsiveContainer
              debounce={0}
              height={90}
              width={115}
            >
              <ResizeDetector
                handleHeight={true}
                handleWidth={true}
                onResize={[Function]}
                targetRef={
                  Object {
                    "current": <div
                      class="recharts-responsive-container"
                      style="width: 115px; height: 90px;"
                    >
                      <div
                        class="recharts-wrapper"
                        style="position: relative; cursor: default; width: 115px; height: 90px;"
                      >
                        <svg
                          class="recharts-surface"
                          height="90"
                          version="1.1"
                          viewBox="0 0 115 90"
                          width="115"
                        >
                          <defs>
                            <clippath
                              id="recharts1-clip"
                            >
                              <rect
                                height="80"
                                width="105"
                                x="5"
                                y="5"
                              />
                            </clippath>
                          </defs>
                          <defs>
                            <lineargradient
                              id="Green"
                              x1="1"
                              x2="0"
                              y1="0"
                              y2="1"
                            >
                              <stop
                                offset="0%"
                                stop-color="#2aac63"
                                stop-opacity="0.17"
                              />
                              <stop
                                offset="100%"
                                stop-color="#2aac63"
                                stop-opacity="0"
                              />
                            </lineargradient>
                            <lineargradient
                              id="White"
                              x1="0"
                              x2="0"
                              y1="0"
                              y2="1"
                            >
                              <stop
                                offset="0%"
                                stop-color="#fff"
                                stop-opacity="0"
                              />
                              <stop
                                offset="100%"
                                stop-color="#fff"
                                stop-opacity="0"
                              />
                            </lineargradient>
                            <lineargradient
                              id="Blue"
                              x1="1"
                              x2="0"
                              y1="0"
                              y2="1"
                            >
                              <stop
                                offset="0%"
                                stop-color="#4689b6"
                                stop-opacity="0.17"
                              />
                              <stop
                                offset="100%"
                                stop-color="#4689b6"
                                stop-opacity="0"
                              />
                            </lineargradient>
                            <lineargradient
                              id="Pink"
                              x1="1"
                              x2="0"
                              y1="0"
                              y2="1"
                            >
                              <stop
                                offset="0%"
                                stop-color="#a075b9"
                                stop-opacity="0.17"
                              />
                              <stop
                                offset="100%"
                                stop-color="#a075b9"
                                stop-opacity="0"
                              />
                            </lineargradient>
                            <lineargradient
                              id="Orange"
                              x1="1"
                              x2="0"
                              y1="0"
                              y2="1"
                            >
                              <stop
                                offset="0%"
                                stop-color="#db9a67"
                                stop-opacity="0.17"
                              />
                              <stop
                                offset="100%"
                                stop-color="#db9a67"
                                stop-opacity="0"
                              />
                            </lineargradient>
                            <lineargradient
                              id="Blue2"
                              x1="1"
                              x2="0"
                              y1="0"
                              y2="1"
                            >
                              <stop
                                offset="0%"
                                stop-color="#4a9cc0"
                                stop-opacity="0.17"
                              />
                              <stop
                                offset="100%"
                                stop-color="#4a9cc0"
                                stop-opacity="0"
                              />
                            </lineargradient>
                          </defs>
                          <g
                            class="recharts-layer recharts-area"
                          >
                            <g
                              class="recharts-layer recharts-area-dots"
                            />
                          </g>
                          <g
                            class="recharts-layer recharts-area"
                          >
                            <g
                              class="recharts-layer recharts-area-dots"
                            />
                          </g>
                          <g
                            class="recharts-layer recharts-area"
                          >
                            <g
                              class="recharts-layer recharts-area-dots"
                            />
                          </g>
                          <g
                            class="recharts-layer recharts-area"
                          >
                            <g
                              class="recharts-layer recharts-area-dots"
                            />
                          </g>
                          <g
                            class="recharts-layer recharts-area"
                          >
                            <g
                              class="recharts-layer recharts-area-dots"
                            />
                          </g>
                        </svg>
                      </div>
                    </div>,
                  }
                }
              >
                <div
                  className="recharts-responsive-container"
                  style={
                    Object {
                      "height": 90,
                      "maxHeight": undefined,
                      "minHeight": undefined,
                      "minWidth": undefined,
                      "width": 115,
                    }
                  }
                />
              </ResizeDetector>
            </ResponsiveContainer>
            <Grid>
              <div
                className="grid"
              >
                <span
                  className="grid_card"
                />
                <span
                  className="grid_card"
                />
                <span
                  className="grid_card"
                />
                <span
                  className="grid_card"
                />
                <span
                  className="grid_card"
                />
              </div>
            </Grid>
          </div>
        </div>
      </Graph>
    </div>
  </div>
</InfoBox>
`;
